lib/repl.js

```diff
@@ -413,18 +409,13 @@ function REPLServer(prompt,
   if (!input && !output) {
     // legacy API, passing a 'stream'/'socket' option
     if (!stream) {
-      // use stdin and stdout as the default streams if none were given
+      // Use stdin and stdout as the default streams if none were given
       stream = process;
     }
-    if (stream.stdin && stream.stdout) {
-      // We're given custom object with 2 streams, or the `process` object
-      input = stream.stdin;
-      output = stream.stdout;
-    } else {
-      // We're given a duplex readable/writable Stream, like a `net.Socket`
-      input = stream;
-      output = stream;
-    }
+    // We're given a duplex readable/writable Stream, like a `net.Socket`
+    // or a custom object with 2 streams, or the `process` object
+    input = stream.stdin || stream;
```

Comment:
It probably doesn't make a difference, but this has the potential to be a behavior change if only one of `stream.stdin` or `stream.stdout` were passed in.

Reply:
Hm, do you want me to change it to the way it was before? I personally do not feel like this is ever going to matter.